src/theory/theory_model.h

```diff
@@ -152,7 +152,18 @@ class TypeSet {
   TypeToTypeEnumMap d_teMap;
   TypeEnumeratorProperties * d_tep;
 
-  public:
+  void addSubTerms(TNode n, std::map< TNode, bool >& visited, bool topLevel=true){
```

Comment:
As with addToTypeList, I'd rather have this function implemented in a non-recursive fashion. But let's decide tomorrow on how to deal with recursive vs non-recursive.